core/trino-main/src/main/java/io/trino/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -550,7 +550,7 @@ protected RowExpression visitSimpleCaseExpression(SimpleCaseExpression node, Voi
 
             arguments.add(node.getDefaultValue()
                     .map(defaultValue -> process(defaultValue, context))
-                    .orElse(constantNull(returnType)));
+                    .orElseGet(() -> constantNull(returnType)));
```

Comment:
(X) Object creation

Reply:
Does it matter that we do a `rnn` check there?

Reply:
reverting